web/src/components/DatePicker.tsx

```diff
@@ -132,34 +89,35 @@ const CustomDatePicker = ({
 
   return (
     <DateContainer>
-      <StyledInputWrapper>
-        <StyledInput
-          open={isCalendarOpen}
-          value={value}
-          popperPlacement='bottom'
-          selected={isValidJsDate(date?.toJSDate() ?? null) ? date?.toJSDate() : null}
-          onSelect={e => {
-            setValue(DateTime.fromJSDate(e ?? new Date()).toFormat('dd.MM.yyyy'))
-            setDatePickerError('')
+      <LocalizationProvider dateAdapter={CustomAdapterLuxon} adapterLocale={languageCode}>
+        <DatePicker
+          value={date}
+          onChange={newValue => {
+            if (newValue) {
+              handleDateChange(newValue.toJSDate())
+              setValue(newValue.toFormat('dd.MM.yyyy'))
+              setDatePickerError('')
+            }
+          }}
+          slotProps={{
+            textField: {
+              placeholder: placeholderDate
+                .setLocale(languageCode)
+                .toLocaleString({ month: '2-digit', day: '2-digit', year: 'numeric' }),
+            },
+            openPickerButton: {
+              'aria-label': calendarLabel,
+            },
+          }}
+          onError={reason => {
+            if (reason) {
+              setDatePickerError(t('invalidDate'))
+            }
           }}
-          onClickOutside={() => setIsCalendarOpen(false)}
-          calendarClassName='calenderStyle'
-          dateFormat='dd.MM.yyyy'
-          placeholderText={placeholderDate.toFormat('dd.MM.yyyy')}
-          onChange={(date: Date | null) => handleDateChange(date)}
-          onChangeRaw={e => handleDateError(String((e?.target as HTMLInputElement).value))}
         />
-        <StyledIconButton
-          color='primary'
-          size='large'
-          name={calendarLabel}
-          isCalendarOpen={isCalendarOpen}
-          onClick={() => setIsCalendarOpen(true)}>
-          <EventIcon />
-        </StyledIconButton>
-      </StyledInputWrapper>
+      </LocalizationProvider>
       <StyledTitle>{title}</StyledTitle>
-      {!!(error || datePickerError) && <StyledError>{error || datePickerError}</StyledError>}
+      {error || datePickerError ? <StyledError>{error || datePickerError}</StyledError> : null}
```

Comment:
🔧 Should we use the built-in/suggested way to display an error instead?
https://mui.com/x/react-date-pickers/validation/#show-the-error